lib/order.rb

```diff
@@ -0,0 +1,108 @@
+require 'csv'
+
+require_relative 'customer.rb'
+
+class Order
+
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  FULLFILLMENT_STATUS = [:pending, :paid, :processing, :shipped, :complete]
+
+  def initialize (id, products, customer, status = :pending)
+    if FULLFILLMENT_STATUS.include?(status) == false
+      return raise ArgumentError, 'Not a status'
+    end
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = status
+
+  end
+
+  def add_product(item, price)
+    if @products.has_key?(item)
+      return raise ArgumentError, 'Item already on list'
+    else
+      new_item = {item => price}
+      @products = @products.merge(new_item)
+    end
+  end
+
+  def remove_product(item)
+    if @products.has_key?(item) == false
+      return raise ArgumentError, 'Item not on list'
+    else
+      @products = @products.delete_if { |key, value| key == item }
+    end
+  end
+
+  def total
+    total = 0.00
+    @products.each_value do |value|
+      total += value
+    end
+    total += (total * 0.075) #7.5% tax
+    return total.round(2)
+  end
+
+  def self.all
+    @orders = []
+    CSV.open("./data/orders.csv", "r").each do |row|
+      order_id = row[0].to_i
+
+      # go through CSV product list for the current customer, split each for processing into a hash to meet initialization requirement
+      order_products = {}
+      raw_products = row[1]
+      raw_products = raw_products.split(';')
+      raw_products.each do |raw_product|
+        raw_product = raw_product.split(':')
+        product = { raw_product[0] =>
+          raw_product[1].to_f }
+          order_products = order_products.merge(product)
+        end
```

Comment:
your indentation is a little off in this `.each` block so it confused me for a second!